src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorApplicationModelProvider.cs

```diff
@@ -71,6 +73,34 @@ public void OnProvidersExecuting(ApplicationModelProviderContext context)
                     AddInvalidModelStateFilter(actionModel);
 
                     InferParameterBindingSources(actionModel);
+
+                    AddMultipartFormDataConsumesAttribute(actionModel);
+                }
+            }
+        }
+
+        // Internal for unit testing
+        internal void AddMultipartFormDataConsumesAttribute(ActionModel actionModel)
+        {
+            if (_apiBehaviorOptions.SuppressAddingConsumesConstraintForFormParameters)
+            {
+                return;
+            }
+
+            // Add a ConsumesAttribute if the request does not explicitly specify one.
+            if (actionModel.Filters.OfType<IConsumesActionConstraint>().Any())
+            {
+                return;
+            }
+
+            foreach (var parameter in actionModel.Parameters)
+            {
+                var bindingSource = parameter.BindingInfo?.BindingSource;
+                if (bindingSource == BindingSource.Form || bindingSource == BindingSource.FormFile)
+                {
+                    // If an action accepts files, it must accept multipart/form-data. This accept header
+                    // must preclude any other 
```

Comment:
This has nothing to do with accept header, this doesn't super make sense

Reply:
Sorry, I meant Content-Type. We essentially prefer the Consumes attribute that's specified last. I was attempting to say we need it at the end so that it overrides any previously configured attribute.